cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -349,6 +374,10 @@ func (pm *PipelineManager) Configure(ctx context.Context) (result *PipelineConfi
 		additionalVariables[environment.ResourceGroupEnvVarName] = rgGroup
 	}
 
+	// Merge azd required variables and secrets with the ones defined on azure.yaml
+	additionalVariables, additionalSecrets = pm.configOptions.SecretsAndVars(
+		additionalVariables, additionalSecrets, pm.env.Dotenv())
```

Comment:
Only for values and not for config?

Reply:
config is automatically set as a secret.